src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(arg) {
+    if (arg === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    } else {
+      sum += arg;
+
+      return adder;
+    }
+  };
```

Comment:
The function `adder` does not handle non-numeric inputs. If a non-numeric value is passed to `adder`, it will result in an unintended behavior when adding to `sum`. Consider adding a check to ensure that `arg` is a number before adding it to `sum`.